tests/test_embed/test_interpreter.cpp

```diff
@@ -169,8 +169,17 @@ TEST_CASE("There can be only one interpreter") {
 }
 
 bool has_pybind11_internals_builtin() {
-    auto builtins = py::handle(PyEval_GetBuiltins());
-    return builtins.contains(PYBIND11_INTERNALS_ID);
+    py::dict state_dict;
```

Comment:
I will rename the function and add a comment that this duplicates code elsewhere. I am hesitant to add it to a header file, since it will just add unused code to other projects.

Reply:
Never mind, I will factor it out into a separate function. With all that extra error checking, it is becoming too large and duplicated otherwise.